src/shims/unix/linux/epoll.rs

```diff
@@ -422,43 +431,115 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         events_op: &OpTy<'tcx>,
         maxevents: &OpTy<'tcx>,
         timeout: &OpTy<'tcx>,
-    ) -> InterpResult<'tcx, Scalar> {
+        dest: MPlaceTy<'tcx>,
```

Comment:
Might be a bit nicer in terms of the interface to pass a reference here, and only clone inside this function when needed

Reply:
Ah I think at first I tried passing the dest place by deference, but I ran into problem of needing to propagate the ``tcx`` runtime. Unfortunately the commit is squashed, but the initial problem description is still in the history of PR description. 

----------------------------------------------------------------------------------------------------------------------------------------------------------
This is the summary, I ommited the compiler error: 

This is a work in progress PR opened for discussion purpose.

In https://github.com/rust-lang/miri/commit/f139c7225ae0ce1adf3c42aab905092942203ad8, the reason that 'tcx is annotated in so many place is because I have used this:
```rust
                this.block_thread(
                    BlockReason::Epoll,
                    Some((TimeoutClock::Monotonic, TimeoutAnchor::Relative, duration)),
                    callback!(
                        @capture<'tcx> {
                        epfd_value: i32,
                        place: &'tcx MPlaceTy<'tcx>, // the culprit is here!!
                        event: MPlaceTy<'tcx>,
                        }
                        @unblock = |this| {
                                    this.blocking_epoll_callback(epfd_value, place, &event);
                                    Ok(())
                                }
                        @timeout = |this| {
                        this.write_scalar(Scalar::from_i32(0), place)?;
                                    Ok(())
                                }
                    ),
                );
```

place: &'tcx MPlaceTy<'tcx> is required to be annotated with 'tcx. Since it is passed from shims/foreign_items.rs, every of them need to be annotated with 'tcx too. I managed to follow the compiler's hint and add them one by one until I hit this error:

It seems like if I want to change the parameters in call_extra_fn, I need to go all the way to rustc miri driver to change [this function](https://github.com/rust-lang/rust/blob/9859bf27fd9892f48725c59b56aeee2be1d2fbad/compiler/rustc_const_eval/src/interpret/machine.rs#L217). Is there any way to get around this?

Reply:
Alright, I deleted a reply because I misunderstood what you meant just now.